shell/platform/fuchsia/flutter/vulkan_surface.cc

```diff
@@ -226,57 +228,112 @@ bool VulkanSurface::CreateFences() {
+    const SkISize& size) {
   if (size.isEmpty()) {
     return false;
   }
 
-  VulkanImage vulkan_image;
-  if (!CreateVulkanImage(vulkan_provider_, size, &vulkan_image)) {
-    FML_DLOG(ERROR) << "Failed to create VkImage";
+  fuchsia::sysmem::BufferCollectionTokenSyncPtr local_token;
+  zx_status_t status =
+      sysmem_allocator->AllocateSharedCollection(local_token.NewRequest());
+  LOG_AND_RETURN(status != ZX_OK, "Failed to allocate collection");
+  fuchsia::sysmem::BufferCollectionTokenSyncPtr scenic_token;
+  status = local_token->Duplicate(std::numeric_limits<uint32_t>::max(),
+                                  scenic_token.NewRequest());
+  LOG_AND_RETURN(status != ZX_OK, "Failed to duplicate token");
+  status = local_token->Sync();
+  LOG_AND_RETURN(status != ZX_OK, "Failed to sync token");
+  fuchsia::sysmem::BufferCollectionTokenSyncPtr vulkan_token;
+  status = local_token->Duplicate(std::numeric_limits<uint32_t>::max(),
+                                  vulkan_token.NewRequest());
+  LOG_AND_RETURN(status != ZX_OK, "Failed to duplicate token");
+  status = local_token->Sync();
+  LOG_AND_RETURN(status != ZX_OK, "Failed to sync token");
+
+  session_->RegisterBufferCollection(buffer_id_, std::move(scenic_token));
+
+  fuchsia::sysmem::BufferCollectionSyncPtr buffer_collection;
+  status = sysmem_allocator->BindSharedCollection(
+      std::move(local_token), buffer_collection.NewRequest());
+  LOG_AND_RETURN(status != ZX_OK, "Failed to bind collection");
+
+  fuchsia::sysmem::BufferCollectionConstraints constraints;
+  constraints.min_buffer_count = 1;
+  constraints.usage.vulkan =
+      fuchsia::sysmem::VULKAN_IMAGE_USAGE_COLOR_ATTACHMENT |
+      fuchsia::sysmem::VULKAN_IMAGE_USAGE_TRANSFER_DST |
+      fuchsia::sysmem::VULKAN_IMAGE_USAGE_TRANSFER_SRC |
+      fuchsia::sysmem::VULKAN_IMAGE_USAGE_SAMPLED;
+
+  constraints.image_format_constraints_count = 1;
+  fuchsia::sysmem::ImageFormatConstraints& image_constraints =
+      constraints.image_format_constraints[0];
+  image_constraints = fuchsia::sysmem::ImageFormatConstraints();
+  image_constraints.min_coded_width = size.width();
+  image_constraints.min_coded_height = size.height();
+  image_constraints.max_coded_width = size.width();
+  image_constraints.max_coded_height = size.height();
+  image_constraints.min_bytes_per_row = 0;
+  image_constraints.pixel_format.type = kSysmemPixelFormat;
+  image_constraints.color_spaces_count = 1;
+  image_constraints.color_space[0].type = fuchsia::sysmem::ColorSpaceType::SRGB;
+
+  // Force bytes per row to 4 * |width_| when using linear buffer.
+  if (vulkan_image_.vk_image_create_info.tiling == VK_IMAGE_TILING_LINEAR) {
+    image_constraints.pixel_format.has_format_modifier = true;
+    image_constraints.pixel_format.format_modifier.value =
+        fuchsia::sysmem::FORMAT_MODIFIER_LINEAR;
+    image_constraints.min_bytes_per_row = size.width() * 4;
+    image_constraints.max_bytes_per_row = size.width() * 4;
+  }
```

Comment:
why is this needed? failing to see when tiling will ever be set to LINEAR and even then I'm not sure why we want this..